includes/class-boldgrid-backup-archive-fetcher.php

```diff
@@ -0,0 +1,290 @@
+
+		$this->response = wp_remote_get(
+			$this->url,
+			[
+				'filename'  => $this->filepath,
+				'headers'   => 'Accept: ' . implode( ', ', $this->allowed_content_types ),
+				'sslverify' => false,
+				'stream'    => true,
+				'timeout'   => MINUTE_IN_SECONDS * 20,
+			]
+		);
+
+		if ( $this->is_successful_download() ) {
+			$this->post_successful_download();
+
+			return true;
+		} else {
+			$this->core->wp_filesystem->delete( $this->filepath );
+
+			$this->error = __(
+				'Could not retrieve the remote file.  It may not be a ZIP file, or the link is no longer valid.',
+				'boldgrid-backup'
+			);
+
+			return false;
+		}
+	}
+
+	/**
+	 * Get our error message.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @return string
+	 */
+	public function get_error() {
+		return $this->error;
+	}
+
+	/**
+	 * Get our array of info after a successful download.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @return array
+	 */
+	public function get_info() {
+		return $this->info;
+	}
+
+	/**
+	 * Whether or not we encountered an error during the download process.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @return bool
+	 */
+	public function has_error() {
+		return ! empty( $this->error );
+	}
+
+	/**
+	 * Whether or not the call to download the file was successful.
+	 *
+	 * This does not represent the success of the download() method, but instead the state of the
+	 * wp_remote_get call.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @return bool True on success.
+	 */
+	private function is_successful_download() {
+		return is_array( $this->response ) &&
+			! is_wp_error( $this->response ) &&
+			in_array( $this->response['headers']['content-type'], $this->allowed_content_types, true );
+	}
+
+	/**
+	 * Validate our backup directory.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @return bool True if valid.
+	 */
+	private function is_valid_backupdir() {
+		$backup_directory = $this->core->backup_dir->get();
+
+		return $this->core->backup_dir->is_valid( $backup_directory ) && empty( $this->core->backup_dir->errors );
+	}
+
+	/**
+	 * Validate our download url.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @return bool True if valid.
+	 */
+	private function is_valid_url() {
+		$url_regex = '/' . $this->core->configs['url_regex'] . '/i';
+
+		return preg_match( $url_regex, $this->url );
+	}
+
+	/**
+	 * Steps to take if our raw wp_remote_get() call to download the backup was successful.
+	 *
+	 * @since SINCEVERSION
+	 */
+	private function post_successful_download() {
+		$this->set_logfilepath();
+
+		$filename = basename( $this->filepath );
+
+		// Restore the log file from the archive.
+		$this->core->archive_log->restore_by_zip( $this->filepath, basename( $this->log_filepath ) );
+
+		// Update the archive file modification time, based on the log file contents.
+		$this->core->remote->post_download( $this->filepath );
+
+		// Get the archive details.
+		$archive = $this->core->archive->get_by_name( $filename );
+
+		$this->info = [
```

Comment:
Arrays should be declared with long array syntax per WordPress coding standards
https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#declaring-arrays

Reply:
See similar comment above.